### DISCUSSION TITLE
I can't generate Purview Reports

### DISCUSSION THREAD

**Discussion by vanessav5** [silverhack/monkey365#127] [General]
Hi,

I have executed Monkey and I can obtain all reports, but I have the following error when I try to generate Purview Reports (I have assigned license)

PS C:\monkey365> $assets =  Invoke-Monkey365 -Export HTML -PromptBehavior SelectAccount -IncludeEntraID -Instance Microsoft365 -Collect Purview
WARNING: [] - [Initialize-MonkeyScan] - Collectors were not found for Microsoft365 - warning -
DESKTOP - Monkey365InitializeScan
Write-Error : [] - [Invoke-MonkeyScanner] - The property 'libCommands' cannot be found on this object.
Verify that the property exists.. LineNumber: At C:\monkey365\core\tasks\Invoke-MonkeyScanner.ps1:128 char:13
+             $libs = $all_scans.libCommands | Select-Object -Unique
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ - exception - DESKTOP -
At C:\monkey365\core\tasks\Invoke-MonkeyScanner.ps1:220 char:13
+             Write-Error $_
+             ~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Write-Error

Could you help me please with this error?.  I'd like to review my Purview .

Thanks and regards,

Vanessa

**Comment by silverhack** [comment]
Hi @vanessav5 

Apparently it's a bug. I can reproduce your issue and seem to appear only with purview service. I'll let you know as soon as I can. 

Thanks for letting me know about that.

**Reply by vanessav5** [reply]
> Hi @vanessav5
> 
> Apparently it's a bug. I can reproduce your issue and seem to appear only with purview service. I'll let you know as soon as I can.
> 
> Thanks for letting me know about that.

Ok, @silverhack .  Thank you.  I'll be waiting your answer.

Regards,


VAnessa

**Comment by silverhack** [comment]
Hi @vanessav5, 

I can confirm that it's a bug and now it's fixed on [dev branch](https://github.com/silverhack/monkey365/tree/develop). Could you please check and do let me know? 

Cheers,

**Reply by vanessav5** [reply]
Hi @silverhack,

I downloaded the zip from dev branch and I'll try to obtain the report, but I have this error:

PS C:\monkey365> $assets =  Invoke-Monkey365 -Export HTML -PromptBehavior SelectAccount -IncludeEntraID -Instance Microsoft365 -Collect Purview
WARNING: [] - [Get-MonkeyMSALToken] - Acquire token failed. An error occurred while sending the request. -
warning - DESKTOP-79CFM6G -


I'll wait for news.

Regards,


VAnessa

**Reply by silverhack** [reply]
Could you please try again? This error is indicating you that the tool was unable to request the endpoint for whatever reason (e.g internet connection error, domain unavailable, etc..)

**Reply by vanessav5** [reply]
> Could you please try again? This error is indicating you that the tool was unable to request the endpoint for whatever reason (e.g internet connection error, domain unavailable, etc..)

Hi @silverhack 

I have executed again but the error continues.

PS C:\WINDOWS\system32> $assets = Invoke-Monkey365 -Export HTML -PromptBehavior SelectAccount -IncludeEntraID -Instance Microsoft365 -Collect Purview
WARNING: [09:49:50:478] - [Get-MonkeyMSALToken] - Acquire token failed. An error occurred while sending the request. -
warning - DESKTOP-79CFM6G -
PS C:\WINDOWS\system32> $assets = Invoke-Monkey365 -Export HTML -PromptBehavior SelectAccount -IncludeEntraID -Instance Microsoft365 -Collect AdminPortal
WARNING: [09:50:06:573] - [Get-MonkeyMSALToken] - Acquire token failed. An error occurred while sending the request. -
warning - DESKTOP-79CFM6G -
PS C:\WINDOWS\system32> $assets = Invoke-Monkey365 -Export HTML -PromptBehavior SelectAccount -IncludeEntraID -Instance Microsoft365 -Collect MicrosoftTeams
WARNING: [09:50:15:414] - [Get-MonkeyMSALToken] - Acquire token failed. An error occurred while sending the request. -
warning - DESKTOP-79CFM6G -
PS C:\WINDOWS\system32>


And I executed the main package and I could..

Maybe is the develop package.. 


I hope your answer


Vanessa

**Reply by vanessav5** [reply]
I can't execute Monkey for any collect :( ..I have internet in my pc but nothing.  What is the error now?

Regards,

Vanessa

**Reply by vanessav5** [reply]
Hi @silverhack,

I can execute the report.  With the last development package. 

Regards,

Vanessa

**Reply by silverhack** [reply]
Hi @vanessav5 I will have a look into the dev branch and will let you know as soon as I can
